### ISSUE TITLE
Data/Media/Informational Resources

### ISSUE THREAD

**Issue by gcassel** [valueflows/valueflows#161]
I'm finally at a controlled "request for comments" phase with the development of my [P2P Digital Networking Protocol](https://docs.google.com/document/d/1CKXns9XKySsFGcPGnmKiM5wAWJmbeBWZY6iIljAz6a4/edit?usp=sharing) document.   

That document only relates partially to VF's goals.  However, it relates deeply to my personal focus on the immateriality of media/ informational resources.  As I've occasionally (briefly) mentioned, I believe that media resources should be fundamentally distinguished from physical resources.

**I'd appreciate ideas on how to distinguish digital media from physical resources in vf accounting.**  I'm inclined to simply suggest *vf:DataResource*, to indicate any valuable combination of data or media which networks may wish to account for in economic processes (separately from physical copies.)

**Comment by fosterlynn** [comment]
There has been some inconclusive discussion here: https://github.com/valueflows/resource/issues/53

MaterialResource vs DigitalResource and/or KnowledgeResource.....

**Comment by bhaugen** [comment]
I think we need to distinguish several high-level resource types that have different behaviors. Digital resources being one of them, money another, perishables yet another. Some of this typification might need to be faceted, like perishables will not be the only resources that are lot-controlled.

Re digital resources: @gcassel do you think that media resources like music and video will require significantly different treatment than knowledge resources in digital document form?

**Comment by fosterlynn** [comment]
...and there was discussion earlier around the particular needs of media, where people might use snippets of a video or whatever and want credit for it (we knew a couple people trying to do media as an OVN, but I think they are both stalled out).....

But yes all a good topic, and one that I think will take some fairly deep discussion and many examples.

**Comment by gcassel** [comment]
> I think we need to distinguish several high-level resource types that have different behaviors. Digital resources being one of them, money another, perishables yet another. 

^ Agreed.

**Comment by gcassel** [comment]
> @gcassel do you think that media resources like music and video will require significantly different treatment than knowledge resources in digital document form?

Do you specifically mean the difference between representations of music, video, images etcetera and _text_?

Language, including written text, is fundamentally different from (but not mutually exclusive of) non-language representations of visual and acoustic information.   However, I think that all media resources can be handled according to some generic participation principles, including subjects such as creative credits, licensing restrictions, {editing, viewing and commenting rights}, etcetera.   

Some creative and collaborative distinctions do relate to the basic differences between media types.    I haven't looked closely at that as a subject for p2p networking tools and techniques, as of yet.

**Comment by gcassel** [comment]
> ...and there was discussion earlier around the particular needs of media, where people might use snippets of a video or whatever and want credit for it

As you may know @fosterlynn , it's possible to reference specific sections/ selections of streaming videos in URLs, with [tools like this](http://youtubetime.com/).  That principle can easily be adapted for RDF formatting of video selections of streaming videos _and_ of video files, enabling selection-based sharing, commenting, evaluating etcetera.     (This is one of my main media networking goals.)

**Comment by bhaugen** [comment]
> > @gcassel do you think that media resources like music and video will require significantly different treatment than knowledge resources in digital document form?
> 
> Do you specifically mean the difference between representations of music, video, images etcetera and _text_?

No, I mean socially/economically music especially seems to have a bunch of rules/customs/expectations and even economic networks around it. I expect video does too. Likewise visual art. 

Scientific documents have their own sets of all of that stuff, e.g. [provenance](https://github.com/valueflows/valueflows/wiki/Provenance) and well-developed citation behaviors. Provenance does not apply only to scientific documents, but it's well-developed there.

Does that help focus in on my question?

**Comment by fosterlynn** [comment]
> As you may know @fosterlynn , it's possible to reference specific sections/ selections of streaming videos in URLs

Yes I perceptually know that, I'm sure I don't know the in's and out's and all the implications in the sharing, citing, etc of those.  But yes.

What I definitely don't know is how to best model that in terms of vf:Resources.  The video is a resource; the section of the video is a resource; but there are infinite (well a whole lot of) possibilities of how to take a slice a video, so it isn't a thing where you can say these 10 slices make up the video.....  

At some point, I'd like to understand how people who make videos (and music, etc.) think of these things, what the practices are, especially for people who want to do it as commons on some level.

**Comment by gcassel** [comment]
I understand your questions now @bhaugen and @fosterlynn to be related mainly to the difference between science and art.   Does that seem accurate, or too reductive?   

> ...socially/economically music especially seems to have a bunch of rules/customs/expectations and even economic networks around it. I expect video does too. Likewise visual art.

It varies wildly within and between art disciplines, and according to the collaborative scale and industrialization of artistic production processes. 

Creative and technical roles are sometimes neatly distinguished; other times they aren't.  Certainly, there's broad standardization of the _types_ and the _level of detail_ of creative _and_ technical credits for some massively collaborative arts, such as motion picture production.   

The importance of creative credits and of intellectual property is extremely controversial in many artistic fields, such as music, video and their [remix](https://en.wikipedia.org/wiki/Remix) cultures.   Openly borrowing and remixing others' creations is more normalized in, say, cuisine and fashion design.  

I guess @bhaugen my answer is that no, I don't expect that art products will require significantly different treatment from knowledge resources in vf accounting-- unless vf eventually tries to deal with issues such as the reliability and trustworthiness of knowledge resources.  Those aspects don't really apply to art.   However, the production of all art and knowledge media resources involves unpredictable creative activity, in which it's hard to accurately _describe_ and _proportionally evaluate_ specific contributions to the whole.

**Comment by bhaugen** [comment]
> unless vf eventually tries to deal with issues such as the reliability and trustworthiness of knowledge resources. 

I don't expect us to do do.

**Comment by bhaugen** [comment]
Does this (ActivityPub) relate to this VF issue?
- http://mediagoblin.org/news/tpac-2016-and-review-activitypub.html
- https://www.w3.org/TR/activitypub/

It's from http://mediagoblin.org/ and I am not totally sure how it relates to media, but I know that is Christopher Allan Webber's focus. He's a an old [ChiPy](http://www.chipy.org/) guy, as am I.

**Comment by gcassel** [comment]
I think ActivityPub relates to my very broad term of "media resources", but not in a very generic way... as far as I can tell from a quick look, it's specifically about reporting/ documenting:

> In the most basic sense, an "Activity" is a semantic description of an action. 
> (from https://www.w3.org/TR/activitystreams-core/) 

I want to distinguish all data/media/informational resources (and signals!) from physical resources, as a start.   Description/ classification of informational resources only matters IMO if that distinction is recognized.  

I think VF does need to look really closely at stuff such as this W3C conception of "Activity", along the lines with VF's concerns about events/  observations/ reports.    And frankly, I'd rather if VF used "action", "interaction" and/or "activity" than "event" for such things, because "event" is a very deeply useful term for social organization and planning of activities.   Of course I realize the REA model is Resource Event Agent, but maybe we (or at least some of us) should be thinking instead of Resource Action Agent.

**Comment by bhaugen** [comment]
@gcassel 

> I want to distinguish all data/media/informational resources (and signals!) from physical resources, as a start. Description/ classification of informational resources only matters IMO if that distinction is recognized.

I sense some worry. Do you think there is any part of that territory that all of us here don't agree with?

How should we move forward on your concerns?

**Comment by gcassel** [comment]
I'm not worried about our agreement on the basic issues here, @bhaugen .  I'm somewhat concerned about our ability to _find time_ to progress on this, in addition to the other big subjects in VF.  I realize however that folks here are easy to work with, and oriented towards efficient progress.

I mentioned the idea of a _vf:DataResource_ in the OP here, but per @elf-pavlik 's comments in [valueflows/resource#53](https://github.com/valueflows/resource/issues/53) which @fosterlynn linked above (thanks),  I think _vf:KnowledgeResource_ would be better.   Either _KnowledgeResource_ or _InfoResource_ I guess.   

My draft p2p digital networking protocol has a [Resource Description Frames](https://docs.google.com/document/d/1CKXns9XKySsFGcPGnmKiM5wAWJmbeBWZY6iIljAz6a4/edit#heading=h.vlwkk4cu8sg1) section, which is IMO way too detailed for VF,  but which does indicate some potentially useful distinctions for VF accounting, especially  Language(s), Format(s), License/Restrictions and Credits.   I believe those subjects are highly relevant to knowledge resources in general.   So, I suggest those could be good descriptive elements/frames for identified knowledge resources in VF accounting.    Any thoughts?

**Comment by bhaugen** [comment]
> I mentioned the idea of a vf:DataResource in the OP here, but per @elf-pavlik 's comments in valueflows/resource#53 which @fosterlynn linked above (thanks), I think vf:KnowledgeResource would be better. Either KnowledgeResource or InfoResource I guess.

I think we need a taxonomy of whatever we end up calling REA:EconomicResourceType in VF. It's like all of the other REA primitives: there are many advantages of using all EconomicResources and their Types with the same conceptual references. What I mean by that is for example whatever we end up calling REA:EconomicEvent will have a reference to a resource type and possibly a resource. The references to a resource type should be at the broadest level of a taxonomy the same concept. Likewise the references to an economic resource. Likewise the references from other concepts to an economic event.

If we can do that, all of the economic flows in the world can be interconnected. If we divorce them from each other - by declaring that a data or knowledge or info resource is divorced as a resource from all other resources - then we start breaking up the interconnections.  

So I agree that data/knowledge/info resources have specialized behavior, but they are also economic resources. And many other types of resources also have specialized behavior, but are also economic resources: e.g. money, perishables, tools, human labor power, etc etc.

Does that all make sense to everybody? Or not? I plan to explore the interconnectedness of all economic life in my next stage of personal work.

**Comment by gcassel** [comment]
> So I agree that data/knowledge/info resources have specialized behavior, but they are also economic resources. 

I agree with that, and with distinguishing knowledge/info/whatever as a type of EconomicResource .   That's what matters IMO within the context of VF's intentions.

**Comment by elf-pavlik** [comment]
I didin't read the whole converstation yet, will try to catch up soon!

One note - similar as knowledge resources which have many representations/interfaces to access them. Similar media resources will have various encodings representing them eg. different codecs and different resolutions 
![youtube-quality-selector](https://www.androidcentral.com/sites/androidcentral.com/files/styles/medium/public/article_images/2014/07/YouTubeQuality3.jpg?itok=5fkdaOQp)

When we want to attribute or review particular media resource, we may want to reference it in consistent way independently of all the possible encoded files available for accessing it.

**Comment by elf-pavlik** [comment]
I didin't read the whole converstation yet, will try to catch up soon!

One note - similar as knowledge resources which have many representations/interfaces to access them. Similar media resources will have various encodings representing them eg. different codecs and different resolutions 
![youtube-quality-selector](https://www.androidcentral.com/sites/androidcentral.com/files/styles/medium/public/article_images/2014/07/YouTubeQuality3.jpg?itok=5fkdaOQp)

When we want to attribute or review particular media resource, we may want to reference it in consistent way independently of all the possible encoded files available for accessing it.

**Comment by gcassel** [comment]
> Similar media resources will have various encodings representing them eg. different codecs and different resolutions 

That's a really excellent point @elf-pavlik  which I hadn't recently considered in distinguishing informational resources from _files_.    That's kind of embarrassing, but I've been juggling lots as I try to integrate distributed file hosting with the inherently "virtual" concept of "community media resources"  (which are _always_ knowledge/informational resources).   

"Format" can apply to media sampling and compressed encoding such as JPG etc...  however, I also use the term "format" to refer to the formatting of spreadsheets, databases etcetera.   

Would it seem accurate to say that the cell-based formatting of a spreadsheet is an inherent part of that knowledge resource,  but the digital encoding used to store a _representation_ of a knowledge resource _in a specific storage location_ is not part of that knowledge resource?

**Comment by fosterlynn** [comment]
This is very interesting. ^

Would this be a similar problem to how to define (for example) food resources when you have carrots, but you also need "carrots in 5 lb bags", "carrots by 1/2 bushel case", "carrots #2's", etc. etc.?  Or is there something here intrinsic to media or knowledge resources?

**Comment by gcassel** [comment]
@fosterlynn "carrots in 5 lb bags" etc is (slightly) complex information, and of course must be represented with complex data.   I disagree with W3C on some things, but I believe that [semantic triples](https://en.wikipedia.org/wiki/Semantic_triple) are the fundamentally sound basis for representing complex data.   Semantic triples can be used with very complex subjects, predicates and objects to create complex associations.   However, they are IMO the fundamentally sound basis for describing even the simplest perceived relationships, including relationships between agents and resources.    

Each subject, predicate and object in a simple semantic triple may each be any kind of scalar variable such as a word or a number, and are inevitably associated with language forms.   I.e. "1024" means one thousand and twenty-four as a _decimal number_, not 1 and 0 and 2 and 4,  and "word" means word in English, not w and o and r and d.

Of course, spreadsheet and database formats are neither semantic triples nor are they natural language, but a great deal of information is currently stored only in spreadsheets and databases.    So, it's very important IMO that Value Flows enables people to describe knowledge resources which exist in such formats.

**Comment by fosterlynn** [comment]
> @fosterlynn "carrots in 5 lb bags" etc is (slightly) complex information, and of course must be represented with complex data.

Well you can order them at that level, or more accurately something like "carrots, 10 - 5 lb bags".  I'm pretty sure that is one "resource model" or "resource type" on a list, even if there is other more complex information around it.  (I have seen exactly that when I load our buying club catalog every month.)

I'm trying to determine what is unique about media / knowledge resources, say "xyz.png" and "xyz.jpg", or "song1 on CD" and "song1 on vinyl", or the @elf-pavlik excellent example above.  Not saying they aren't, just want to understand.  We haven't addressed the question around carrots yet, either. :)  And I'm not sure how far VF wants to go with that level of definition of resources, whether knowledge or material.  Again, not saying it doesn't, just that it is worth thinking about.

**Comment by gcassel** [comment]
> I'm trying to determine what is unique about media / knowledge resources, say "xyz.png" and "xyz.jpg", or "song1 on CD" and "song1 on vinyl", or the @elf-pavlik excellent example above. 

Here's one of the early bits in my current p2p digital networking protocol draft:

> Data can signify all types of information, but typically _represents_ written language, visual or acoustic information.  Such representational data is often called **media**.  

I believe this is true of almost anything which people tend to call media:  it _represents_ written language, visual and/or acoustic information.   However, written language is deeply different (and often much less detailed) than visual and acoustic information.  

Ideas, language and knowledge are all, I believe, essentially intangible or virtual.   Likewise, I believe that all _informational/ knowledge resources_ are intangible or virtual.   For instance, the fact that 2+2=4 is utterly independent of whether or not anyone writes that down, much less whether or not it's written on any _specific_ piece of paper or stored in any _specific_ files.    2+2=4 is a combination of related ideas, which people can express in different languages and represent in various media forms and formats.    

However, many combinations of related ideas are much more distinctive.  For instance:, 
1.  each Shakespearian sonnet is recognizably different from _every_ other combination of ideas, and most if not all of that distinctiveness can be translated into languages other than English.
2.  an original van Gogh painting contains information, but its physical media (canvas and paint) are so distinctive and expressively detailed that information is _substantially lost_ in a photograph or print.   However, a reproduction or representation of a van Gogh painting- including a compressed jpg image-- is certainly a knowledge resource.   It's just a lesser knowledge resource than the original artifact, even disregarding the emotion and historical value we give to artifacts.

Anyway, **whether or not a specific knowledge resource is valuable is IMO utterly dependent on social processes, mutual understandings and agreements**.  There's no way around the social and subjective aspects of it.   Obviously though, it's a big legal issue through intellectual property laws, which coercively restrict the distribution and use of some combinations of ideas. 

Unfortunately, here's where this gets really complicated.

One of the huge underlying issues here is that specific knowledge resources relate to both _types of form_ and to _specific instances of forms_.  For instance, a human is a type of animal, and I'm a specific instance of a human.  This distinction is relevant here because the knowledge resources involved in economic interactions are often reports/observations regarding specific instances of forms.  For instance, all **samples** of information (including the physical information in a rock sample) and all **experiments** create data regarding _specific instances_ of defined _types of form_.   

Of course, I understand that VF folks understand the difference between _types of form_ and _specific instances of form_, and have effectively distinguished that in data models.   However, determining the nature (and value) of complex ideas and knowledge resources really is deeply difficult-- partly because of the casual ways which most people interlace ideas regarding types and instances of form.   

I certainly do want to identify whatever is useful and practical for VF here.   That's probably going to take lots of time and repeated discussions.

**Comment by gcassel** [comment]
FYI, I'm in my last  _request for comments_ phase for **[Peer-to-Peer Digital Networking: How personal computing and the Internet should work](https://docs.google.com/document/d/1atJMtDpCEzVe7f5EQFSsKndvzObsqsnuN7_hDWfwJ98/edit?usp=sharing).**   That's the intro essay for my developing book on signaling, data and media resources.

**Comment by bhaugen** [comment]
@gcassel what's your deadline? I'd like to make another round of annoying comments, but we are even more swamped than usual until at least this coming weekend.

**Comment by gcassel** [comment]
I don't have any hard deadlines yet @bhaugen , but I may be publishing the intro essay/ blog (the link above) by Friday or Monday depending on P2P Foundation's schedule.   I hope to stabilize more of the big document (small book) by then, but it will continue to evolve indefinitely as a public work-in-progress.

[ Edit: published intro essay:   https://blog.p2pfoundation.net/peer-peer-digital-networking-internet-work/2016/10/28   ]
